Bugfix: Optimize quiz footer

lovaiible opened the pull request:
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context
This PR should fix the issue that is mentioned in https://github.com/ls1intum/Artemis/issues/749

### Steps for Testing
1. Create a quiz exercise with at least 7 questions
2. Start or preview exercise and view quiz footer

### Screenshots
![image](https://user-images.githubusercontent.com/33764166/67167957-60268e80-f39f-11e9-849c-f3e4c09fcdac.png)

![image](https://user-images.githubusercontent.com/33764166/67636895-84b9c380-f8d5-11e9-95e5-f5783f5dffee.png)

krusche commented:
Looks good for desktop, but not very nice on mobile.
Can we optimize the footer that it always fits into max. 3 lines?
No matter how small the display is, which mode is used and which language is used?

lovaiible commented:
> Looks good for desktop, but not very nice on mobile.
> Can we optimize the footer that it always fits into max. 3 lines?
> No matter how small the display is, which mode is used and which language is used?

Hi @krusche, I adapted the footer so that at most 3 lines are shown. If the text or the icons should exceed 3 lines, then the overflowing content would be scrollable.

ungaralex commented on src/main/webapp/app/quiz/participate/quiz.component.html:
```diff
@@ -156,15 +155,26 @@ <h2 [ngSwitch]="mode">
+                    [class.btn-sm]="!deviceService.isDesktop()"
+                    class="btn btn-success"
+                    jhiTranslate="artemisApp.quizExercise.submitted"
+                ></button>
```
```suggestion
                <button
                    id="submit-quiz"
                    *ngIf="!showingResult"
                    [disabled]="submission.submitted || isSubmitting || waitingForQuizStart || remainingTimeSeconds < 0"
                    [class.btn-lg]="deviceService.isDesktop()"
                    [class.btn-sm]="!deviceService.isDesktop()"
                    class="btn btn-success"
                    [jhiTranslate]="submission.submitted ? 'artemisApp.quizExercise.submitted' : 'entity.action.submit'"
                    (click)="onSubmit()"
                ></button>
```

ungaralex commented on src/main/webapp/app/quiz/participate/quiz.component.ts:
```diff
@@ -110,6 +111,7 @@ export class QuizComponent implements OnInit, OnDestroy {
         private jhiAlertService: JhiAlertService,
         private quizSubmissionService: QuizSubmissionService,
         private translateService: TranslateService,
+        public deviceService: DeviceDetectorService,
```
I would try to avoid exposing services to the templates. I.m.o. the best practice would be to have some variable/getter/method exposing the information you need.

lovaiible replied:
That's a good point - I changed that :)

krusche commented:
@lovaiible look better but please remove the scrollbar. I don’t consider this as good practice on mobile devices. Either we make the text shorter or the font smaller in this case.

Remaining time could be time left
And seconds could just be sec here
Or we say “saved a moment ago” or we just say saved. I don’t know if we even need to tell the user when the last save was made

What do you think?

lovaiible commented:
@ungaralex  @krusche thank you for the review! I have applied the feedback accordingly.

On the mobile device I have removed the `saved` hint completely. I think if we remove the time, when the last change was saved, but keep the `saved` label, it would be confusing. So only the submit label will be displayed. I have also changed the labels for `remaining time` to `time left` to save some space. I hope this approach is ok @krusche

krusche commented:
Hmm, I don’t understand why?!? I think it is still valuable information, in particular if the save failed for some reason. The time when the save happened is however not important. I think we should display the 3 states „saved“, „saving...“ or „save failed“ on mobile devices (or a similar short formulation).

So please add this info again on mobile devices. This is particularly helpful if someone has a bad internet connection or if the internet connection changes on the mobile device!

lovaiible commented:
Hi @krusche, I have added the label again (without the timestamp) and now it shows 
`saving answers...` when it's currently saving, `saved` when everything is saved and I have kept the old logic that the hint `unsaved changes` is displayed next to the connection status.
![image](https://user-images.githubusercontent.com/33764166/67679348-28b07700-f989-11e9-8ec9-03a5b93490f1.png)

maxr96 commented on src/main/webapp/app/quiz/participate/quiz.component.html:
```diff
@@ -156,15 +161,18 @@ <h2 [ngSwitch]="mode">
-            ></button>
-            <button *ngIf="submission.submitted && !showingResult" disabled class="btn btn-success btn-lg" jhiTranslate="artemisApp.quizExercise.submitted"></button>
+            <div class="submit-button">
+                <button
```
In my opinion, this is a good candidate for using `jhi-button`. For example, button size in this case you will be able to set with `btnSize` input parameter.

lovaiible commented:
@maxr96 @ungaralex Thanks for your review! I implemented your suggestions. Could you please have a look and approve this PR?
